students/Ben_freeman/lesson09/Mailroom_OO/mailroom_OO.py

```diff
@@ -25,17 +24,21 @@ def donor_donations(self):
         """counts the number of all of the donations of a particular donor"""
         return len(self.donor)
 
-    def __str__(self):
+    def message(self):
         """our default thank you message"""
-        return f"\nThank you {self.name} for your donation of ${self.donor_sum:.2f}, your donations make the work "\
-            f"of the 'American society for taking donations' possible.\n\n""Sincearly,\n\n""A low paid intern\n "
+        return f"\nThank you {self.name} for your donations totaling ${self.donor_sum:.2f}, your donations make the " \
+            f" work of the 'American society for taking donations' possible.\n\n""Sincearly,\n\n""A low paid intern\n "
 
+    def update_donor(self, new_donate):
+        """adds a donation to an existing donors profile"""
+        self.donor.append(new_donate)
 
 class DonorCollection(object):
     """Class to handle all of the methods that have to do with multiple donors at once"""
 
     def __init__(self, donor_list):
-        self.donors = donor_list
+        self.donors = {a.name: a.donor for a in donor_list}
+        self.donors_object = donor_list
```

Comment:
you don't need both self.donors and self.donors_object.

As per my example, just have self.donors which is a dictionary with name as a key and Donor object as a value.

`self.donors = {donor.name: donor for donor in donors_list}`

Reply:
I'm confused how I can make me sorting_function on lines 54-59 use Donor.attribute things like you suggested last time, if I only use a dictionary that strips away (I think) the Objectness of my Donor class objects. Can I use the input from the __init__ def (specifically donor_list) in other methods in the same class without renaming them like I did before? Thank you.

Reply:
Mmm, Okay so its a dictionary that identifies all of my donor objects. I think i misread what your sent in your example, so this makes sense now. Cheers

Reply:
Right - your version is a dictionary with key and values as strings.

My version, keys are name (for easy lookup) and values are Donor objects for you to work with.